packages/schematics/angular/utility/dependencies.ts

```diff
@@ -0,0 +1,153 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+import { JsonAstObject, JsonParseMode, parseJsonAst } from '@angular-devkit/core';
+import { SchematicsException, Tree } from '@angular-devkit/schematics';
+import {
+  appendPropertyInAstObject,
+  findPropertyInAstObject,
+} from './json-utils';
+
+
+const pkgJsonPath = '/package.json';
+export enum NodeDependencyType {
+  Default,
+  Dev,
+  Peer,
+  Optional,
+}
+
+export interface NodeDependency {
+  type: NodeDependencyType;
+  packageName: string;
+  version: string;
+  overwrite?: boolean;
+}
+
+export function addPackageJsonDependencies(tree: Tree, dependencies: NodeDependency[]): void {
+    dependencies
+      .forEach(dependency => addPackageJsonDependency(tree, dependency));
+}
+
+export function addPackageJsonDependency(tree: Tree, dependency: NodeDependency): void {
+  const packageJsonAst = _readPackageJson(tree);
```

Comment:
Reparsing the file every time seems inefficient.  Swapping `addPackageJsonDependencies` with `addPackageJsonDependency` as the primary would avoid the repeat parsing.  `addPackageJsonDependency` would then just be a helper that calls the former with a one element array.
An alternative option would be to create a lower level primitive that also takes the parsed JSON AST and then build the two user facing functions on that.